playwright/utils/create-colony.ts

```diff
@@ -0,0 +1,94 @@
+import { type Page } from '@playwright/test';
+
+export const generateRandomString = () => {
+  return Math.random().toString(36).substring(2, 8);
+};
+
+const generateRandomUsername = () => {
+  return `user_${generateRandomString()}`;
+};
+
+const generateRandomEmail = () => {
+  return `test_${generateRandomString()}@example.com`;
+};
+
+const fillUserProfile = async (page: Page) => {
+  const username = generateRandomUsername();
+  const email = generateRandomEmail();
+
+  await page.getByLabel(/Email address/i).fill(email);
+  await page.getByLabel(/username/i).fill(username);
+
+  await page.getByRole('button', { name: /continue/i }).click();
+};
+
+export const selectWalletAndUserProfile = async (page: Page) => {
+  let isUserConnected = false;
+
+  await page.route('**/graphql', async (route, request) => {
+    const postData = request.postDataJSON(); // Get the request body as JSON
+    if (postData.operationName === 'GetUserByAddress') {
+      // Fetch original response.
+      const response = await route.fetch();
+
+      const jsonResponse = await response.json();
+
+      // Check if the user exists and has a profile
+      const items = jsonResponse.data?.getUserByAddress?.items;
+      if (items && items.length > 0) {
+        isUserConnected = !!items[0]?.profile?.displayName;
+      }
+    }
+
+    route.continue();
+  });
+
+  await page
+    .getByRole('button', { name: /connect wallet/i })
+    .first()
+    .click();
+
+  await page.getByLabel(/I agree to/i).click();
+
+  await page.getByText(/dev wallet 2/i).click();
+
+  await page.getByText(/Local Ganache Instance/i).waitFor({ state: 'visible' });
+
+  const loadingIndicator = page.getByText(/Checking your access.../i);
+  const loadingIndicatorVisible = await loadingIndicator.isVisible();
+
+  if (loadingIndicatorVisible) {
+    await loadingIndicator.waitFor({ state: 'hidden' });
+  }
```

Comment:
good staff 🙌  I like this utils 😄

Reply:
> but do we actually need timeout after pressSequentially ?

I'm afraid the only way to see it is to remove it and run on your and other devs machines. ( in my case neither of them is needed). Let me try to remove it.

Reply:
most probably once we properly build the app and also fix the bug with quick typing and validation I mentioned above, there will be no need for any of the timeouts)